src/cache.rs

```diff
@@ -467,6 +438,43 @@ impl CachedPath {
     }
 }
 
+impl Hash for CachedPath {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.hash.hash(state);
+    }
+}
+
+impl PartialEq for CachedPath {
+    fn eq(&self, other: &Self) -> bool {
+        self.path.as_os_str() == other.path.as_os_str()
+    }
+}
+
+impl Eq for CachedPath {}
+
+struct BorrowedCachedPath<'a> {
+    hash: u64,
+    path: &'a Path,
+}
+
+impl Equivalent<CachedPath> for BorrowedCachedPath<'_> {
+    fn equivalent(&self, other: &CachedPath) -> bool {
+        self.path.as_os_str() == other.path.as_os_str()
+    }
+}
+
+impl Hash for BorrowedCachedPath<'_> {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.hash.hash(state);
```

Comment:
This double hashes.. which is a little unfortunate. Maybe what you actually want is the low-level `HashTable` interface that we discussed, but this isn't too bad for now.

Reply:
Good catch. I've changed it to use `state.write_u64()` instead, since we already have a hash that we can use here.

Reply:
Just realized I haven’t actually looked at the implementation of `Hasher` state… so maybe this ends up doing the same thing as before? 🤔

Reply:
Yeah this is the same. The only way to avoid it would be to use a dummy no-op hasher, or papaya adding the raw API.